core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -524,10 +523,14 @@ private[kafka] abstract class AbstractServerThread(connectionQuotas: ConnectionQ
     if (channel != null) {
       debug(s"Closing connection from ${channel.socket.getRemoteSocketAddress()}")
       connectionQuotas.dec(listenerName, channel.socket.getInetAddress)
-      CoreUtils.swallow(channel.socket().close(), this, Level.ERROR)
-      CoreUtils.swallow(channel.close(), this, Level.ERROR)
+      closeSocket(channel)
     }
   }
+
+  def closeSocket(channel: SocketChannel): Unit = {
```

Comment:
looks like this method could be private?

Reply:
it can be protected, but not private.